src/main/java/org/quiltmc/loader/impl/plugin/quilt/StandardQuiltPlugin.java

```diff
@@ -242,12 +242,44 @@ private ModLoadOption[] scan0(Path root, QuiltLoaderIcon fileIcon, ModLocation l
 		PluginGuiTreeNode guiNode) throws IOException {
 
 		Path qmj = root.resolve("quilt.mod.json");
-		if (!FasterFiles.isRegularFile(qmj)) {
+		Path qmj5 = root.resolve("quilt.mod.json5");
+		Path usedQmj = qmj;
+
+		if (FasterFiles.exists(qmj5)) {
+			boolean enableQmj5 = QuiltLoader.isDevelopmentEnvironment() || Boolean.getBoolean(SystemProperties.ENABLE_QUILT_MOD_JSON5);
+			if (!QuiltLoader.isDevelopmentEnvironment() && !enableQmj5) {
```

Comment:
I don't think you should be able to enable this in production, at all - since it allows mods to require the flag at runtime.

Plus this code reads... a bit strangely? This checks for `isDevEnvironment` twice.

Reply:
Ah, I can remove the property; It was more like, an emergency thing than something people should use, but hm, you do have a point; `-Dloader.development=true` already exists after all